odoo/fields.py

```diff
@@ -3477,21 +3477,21 @@ def read(self, records):
         context.update(self.context)
         comodel = records.env[self.comodel_name].with_context(**context)
         inverse = self.inverse_name
-        inverse_field = comodel._fields[inverse]
-        get_id = (lambda rec: rec.id) if inverse_field.type == 'many2one' else int
-        domain = self.get_domain_list(records) + [(inverse, 'in', records.ids)]
-        lines = comodel.search(domain)
-
-        # group lines by inverse field (without prefetching other fields)
-        group = defaultdict(list)
-        for line in lines.with_context(prefetch_fields=False):
-            # line[inverse] may be a record or an integer
-            group[get_id(line[inverse])].append(line.id)
+        get_id = lambda v: v[0] if isinstance(v, tuple) else v
+
+        group = {
+            get_id(read[inverse]): read['related_ids']
+            for read in comodel.read_group(
```

Comment:
I would use `_read_group` as we don't care about semantic ordering

Reply:
@VincentSchippefilt we **do** care about semantic ordering, but not "group" ordering.
The order of groups is not relevant, indeed.
But the order of lines inside each group is important.